portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,26 +33,46 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  ArrayList<String> sampleMessages = new ArrayList<>();
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
-    String messages = convertToJsonString(sampleMessages);
-    response.setContentType("text/html;");
-    response.getWriter().println(messages);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    ArrayList<Comment> comments = new ArrayList<>();
+    for(Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        long timestamp = (long) entity.getProperty("timestamp");
```

Comment:
is this casting needed? Same for (String) in the line below.

Reply:
yea. it generally generates an error if there is no casting

Reply:
Okay. Thanks.